test/e2e/envvars_test.go

```diff
@@ -81,18 +83,78 @@ func TestEnvVars(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error fetching Route %s: %v", names.Route, err)
 	}
-	domain := route.Status.Domain
 
-	envVarsExpectedOutput := fmt.Sprintf("Here are our env vars service:%s - configuration:%s - revision:%s", names.Service, names.Config, names.Revision)
+	domain := route.Status.Domain + "/envvars/should"
 
-	_, err = pkgTest.WaitForEndpointState(
+	resp, err := pkgTest.WaitForEndpointState(
 		clients.KubeClient,
 		logger,
 		domain,
-		pkgTest.Retrying(pkgTest.MatchesBody(envVarsExpectedOutput), http.StatusNotFound),
+		pkgTest.Retrying(verifyUpStatus, http.StatusNotFound),
 		"EnvVarsServesText",
 		test.ServingFlags.ResolvableDomain)
 	if err != nil {
-		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", names.Route, domain, envVarsExpectedOutput, err)
+		t.Fatalf("Failed before reaching desired state : %v", err)
+	}
+
+	expectedOutputs := map[string]string{
+		"K_SERVICE" : names.Service,
+		"K_CONFIGURATION" : names.Config,
+		"K_REVISION" : names.Revision,
+	}
+	// We are failing even for "SHOULD".
+	if err = verifyResponse(resp, "SHOULD", expectedOutputs); err != nil {
+		t.Error(err)
+	}
+
+	domain = route.Status.Domain + "/envvars/must"
+
+	resp, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		logger,
+		domain,
+		pkgTest.Retrying(verifyUpStatus, http.StatusNotFound),
+		"EnvVarsServesText",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("Failed before reaching desired state : %v", err)
+	}
+
+	expectedOutputs = map[string]string{
+		"PORT" : "8080",
```

Comment:
from conformance perspective, my understanding is that as long as PORT is set, it's conformant, doesn't have to be 8080.

Reply:
The contract says "Ingress containerPort for ingress requests and health checks" and in the test image we create the service exposing ":8080" hence the check verifies the port number as well.

Reply:
Can you add a comment explaining why this MUST be 8080 (i.e. "8080 taken from ..., if you change that, this line also needs to change" or the like)?

Suggestion: maybe use a different port from 8080, to catch cases where the port has been hard-coded to 8080?